src/aosm/azext_aosm/generate_nfd/cnf_nfd_generator.py

```diff
@@ -15,7 +37,374 @@ class CnfNfdGenerator(NFDGenerator):
+            data = json.load(f)
+            schema_data = data["properties"]
+
+        try:
+            final_schema = self.find_deploy_params(
+                values_data, schema_data, {}
+            )
+        except KeyError as e:
+            raise InvalidTemplateError(
+                f"ERROR: Your schema and values for the helm package '{helm_package.name}' do not match. Please fix this and run the command again."
+            ) from e
+
+        return final_schema
+
+    def find_deploy_params(
+        self, nested_dict, schema_nested_dict, final_schema
+    ) -> Dict[Any, Any]:
+        original_schema_nested_dict = schema_nested_dict
+        for k, v in nested_dict.items():
+            # if value is a string and contains deployParameters.
+            if isinstance(v, str) and re.search(DEPLOYMENT_PARAMETER_MAPPING_REGEX, v):
+                
+                # only add the parameter name (e.g. from {deployParameter.zone} only param = zone)
+                param = v.split(".", 1)[1]
+                param = param.split("}", 1)[0]
+
+                # add the schema for k (from the big schema) to the (smaller) schema
+                final_schema.update({param: { "type" : schema_nested_dict["properties"][k]["type"]}})
+
+            # else if value is a (non-empty) dictionary (i.e another layer of nesting)
+            elif hasattr(v, "items") and v.items():
+                # handling schema having properties which doesn't map directly to the values file nesting
+                if "properties" in schema_nested_dict.keys():
+                    schema_nested_dict = schema_nested_dict["properties"][k]
+                else:
+                    schema_nested_dict = schema_nested_dict[k]
+                # recursively call function with values (i.e the nested dictionary)
+                self.find_deploy_params(v, schema_nested_dict, final_schema)
+                # reset the schema dict to its original value (once finished with that level of recursion)
+                schema_nested_dict = original_schema_nested_dict
+
+        return final_schema
+
+    def get_chart_name_and_version(
+        self, helm_package: HelmPackageConfig
+    ) -> Tuple[str, str]:
+        chart = os.path.join(
+            self.tmp_folder_name, helm_package.name, "Chart.yaml"
+        )
+
+        with open(chart, "r", encoding="utf-8") as f:
+            data = yaml.load(f, Loader=yaml.FullLoader)
+            chart_name = data["name"]
+            chart_version = data["version"]
+
+        return (chart_name, chart_version)
+
+    def generate_parmeter_mappings(
+        self, helm_package: HelmPackageConfig
+    ) -> str:
+        # Basically copy the values.mappings.yaml file to the right place.
+        values = os.path.join(
+            self.tmp_folder_name, helm_package.name, "values.mappings.yaml"
+        )
+
+        mappings_folder_path = os.path.join(
+            self.tmp_folder_name, "configMappings"
```

Comment:
as said elsewhere, configMappings should be a constant shared between vnf and cnf